File: app/src/test/java/medlife/com/br/model/UsuarioTest.java
package medlife.com.br.model;

import static org.junit.Assert.*;
import org.junit.Before;
import org.junit.Test;
import com.google.firebase.Timestamp;
import java.util.*;

public class UsuarioTest {
    private Usuario usuario;

    @Before
    public void setUp() {
        usuario = new Usuario();
    }

    @Test
    public void testSetAndGetUid() {
        usuario.setUid("uid123");
        assertEquals("uid123", usuario.getUid());
    }

    @Test
    public void testSetAndGetNome() {
        usuario.setNome("John Doe");
        assertEquals("John Doe", usuario.getNome());
    }

    @Test
    public void testSetAndGetEmail() {
        usuario.setEmail("john@example.com");
        assertEquals("john@example.com", usuario.getEmail());
    }

    @Test
    public void testSetAndGetGenero() {
        usuario.setGenero("Masculino");
        assertEquals("Masculino", usuario.getGenero());
    }

    @Test
    public void testSetAndGetOtherGender() {
        usuario.setOtherGender("Não-binário");
        assertEquals("Não-binário", usuario.getOtherGender());
    }

    @Test
    public void testSetAndGetPhoneNumber() {
        usuario.setPhoneNumber("11999999999");
        assertEquals("11999999999", usuario.getPhoneNumber());
    }

    @Test
    public void testSetAndGetDateOfBirth() {
        usuario.setDateOfBirth("1990-01-01");
        assertEquals("1990-01-01", usuario.getDateOfBirth());
    }

    @Test
    public void testSetAndGetCpf() {
        usuario.setCpf("12345678901");
        assertEquals("12345678901", usuario.getCpf());
    }

    @Test
    public void testSetAndGetEndereco() {
        List<Map<String, Object>> enderecos = new ArrayList<>();
        Map<String, Object> endereco = new HashMap<>();
        endereco.put("street", "Rua Teste");
        endereco.put("city", "São Paulo");
        enderecos.add(endereco);
        usuario.setEndereco(enderecos);
        assertEquals(enderecos, usuario.getEndereco());
    }

    @Test
    public void testSetAndGetOrders() {
        List<Order> orders = new ArrayList<>();
        Order order = new Order();
        order.setOrderId("order1");
        orders.add(order);
        usuario.setOrders(orders);
        assertEquals(orders, usuario.getOrders());
    }

    @Test
    public void testSetAndGetCriadoEm() {
        Timestamp now = Timestamp.now();
        usuario.setCriadoEm(now);
        assertEquals(now, usuario.getCriadoEm());
    }

    @Test
    public void testSetAndGetLastLogin() {
        Timestamp now = Timestamp.now();
        usuario.setLastLogin(now);
        assertEquals(now, usuario.getLastLogin());
    }

    @Test
    public void testSetAndGetEnderecoPrincipal() {
        usuario.setEnderecoPrincipal(1);
        assertEquals(1, usuario.getEnderecoPrincipal());
    }

    @Test
    public void testDefaultConstructor() {
        Usuario newUsuario = new Usuario();
        assertNotNull(newUsuario);
        assertNull(newUsuario.getUid());
        assertNull(newUsuario.getNome());
        assertNull(newUsuario.getEmail());
        assertEquals(0, newUsuario.getEnderecoPrincipal());
    }

    @Test
    public void testParameterizedConstructor() {
        Usuario newUsuario = new Usuario("uid123", "John Doe", "john@example.com");
        assertEquals("uid123", newUsuario.getUid());
        assertEquals("John Doe", newUsuario.getNome());
        assertEquals("john@example.com", newUsuario.getEmail());
    }

    @Test
    public void testParameterizedConstructor_NullValues() {
        Usuario newUsuario = new Usuario(null, null, null);
        assertNull(newUsuario.getUid());
        assertNull(newUsuario.getNome());
        assertNull(newUsuario.getEmail());
    }
} 

File: app/src/test/java/medlife/com/br/model/UsuarioTest.kt
package medlife.com.br.model

import com.google.firebase.Timestamp
import org.junit.Assert.*
import org.junit.Before
import org.junit.Test

class UsuarioTest {
    private lateinit var usuario: Usuario

    @Before
    fun setUp() {
        usuario = Usuario()
    }

    @Test fun testSetAndGetUid() { usuario.uid = "uid123"; assertEquals("uid123", usuario.uid) }
    @Test fun testSetAndGetNome() { usuario.nome = "John Doe"; assertEquals("John Doe", usuario.nome) }
    @Test fun testSetAndGetEmail() { usuario.email = "john@example.com"; assertEquals("john@example.com", usuario.email) }
    @Test fun testSetAndGetGenero() { usuario.genero = "Masculino"; assertEquals("Masculino", usuario.genero) }
    @Test fun testSetAndGetOtherGender() { usuario.otherGender = "Não-binário"; assertEquals("Não-binário", usuario.otherGender) }
    @Test fun testSetAndGetPhoneNumber() { usuario.phoneNumber = "11999999999"; assertEquals("11999999999", usuario.phoneNumber) }
    @Test fun testSetAndGetDateOfBirth() { usuario.dateOfBirth = "1990-01-01"; assertEquals("1990-01-01", usuario.dateOfBirth) }
    @Test fun testSetAndGetCpf() { usuario.cpf = "12345678901"; assertEquals("12345678901", usuario.cpf) }

    @Test
    fun testSetAndGetEndereco() {
        val enderecos: List<Map<String, Any>> = listOf(mapOf("street" to "Rua Teste", "city" to "São Paulo"))
        usuario.endereco = enderecos
        assertEquals(enderecos, usuario.endereco)
    }

    @Test
    fun testSetAndGetOrders() {
        val orders = listOf(Order().apply { orderId = "order1" })
        usuario.orders = orders
        assertEquals(orders, usuario.orders)
    }

    @Test
    fun testSetAndGetCriadoEm() {
        val now = Timestamp.now()
        usuario.criadoEm = now
        assertEquals(now, usuario.criadoEm)
    }

    @Test
    fun testSetAndGetLastLogin() {
        val now = Timestamp.now()
        usuario.lastLogin = now
        assertEquals(now, usuario.lastLogin)
    }

    @Test
    fun testSetAndGetEnderecoPrincipal() {
        usuario.enderecoPrincipal = 1
        assertEquals(1, usuario.enderecoPrincipal)
    }

    @Test
    fun testDefaultConstructor() {
        val newUsuario = Usuario()
        assertNotNull(newUsuario)
        assertNull(newUsuario.uid)
        assertNull(newUsuario.nome)
        assertNull(newUsuario.email)
        assertEquals(0, newUsuario.enderecoPrincipal)
    }

    @Test
    fun testParameterizedConstructor() {
        val newUsuario = Usuario().apply {
            uid = "uid123"
            nome = "John Doe"
            email = "john@example.com"
        }
        assertEquals("uid123", newUsuario.uid)
        assertEquals("John Doe", newUsuario.nome)
        assertEquals("john@example.com", newUsuario.email)
    }

    @Test
    fun testParameterizedConstructor_NullValues() {
        val newUsuario = Usuario().apply {
            uid = null
            nome = null
            email = null
        }
        assertNull(newUsuario.uid)
        assertNull(newUsuario.nome)
        assertNull(newUsuario.email)
    }
}
